gobblin-api/src/main/java/gobblin/util/RecordCountProvider.java

```diff
@@ -12,11 +12,34 @@
 
 package gobblin.util;
 
+import org.apache.commons.lang.NotImplementedException;
 import org.apache.hadoop.fs.Path;
 
+
 /**
- * Get record count from a given {@link Path}.
+ * A class for obtaining record counts from data.
  */
-public interface RecordCountProvider {
-  public long getRecordCount(Path path);
+public abstract class RecordCountProvider {
+
+  /**
+   * Get record count from a given {@link Path}.
+   */
+  public abstract long getRecordCount(Path path);
+
+  /**
+   * Convert a {@link Path} so that it can be used in {@link #getRecordCount(Path)} of another
+   * {@link RecordCountProvider}.
+   */
+  public Path convertPath(Path path, RecordCountProvider target) {
```

Comment:
nit: Does it make more sense if changing this to convertPath(path, src)? So the output can be directly used in getRecordCount.